In [1]:
import gspread
import pandas as pd

In [2]:
# load pre-sliced Oncologist and Pthologist data
datadir = '/Users/xthomas/Documents/data/data.cms.gov_Medicare-Physician-Supplier/sliced/'
oncpath12 = 'oncpath12.csv'
oncpath13 = 'oncpath13.csv'
oncpath14 = 'oncpath14.csv'
oncpath15 = 'oncpath15.csv'

In [3]:
df12 = pd.read_csv(datadir + oncpath12)
df13 = pd.read_csv(datadir + oncpath13)
df14 = pd.read_csv(datadir + oncpath14)
df15 = pd.read_csv(datadir + oncpath15)

/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [19]:
df15.head(10)

Unnamed: 0         npi nppes_provider_last_org_name  \
0          12  1003000134                       CIBULL   
1          13  1003000134                       CIBULL   
2          14  1003000134                       CIBULL   
3          15  1003000134                       CIBULL   
4          16  1003000134                       CIBULL   
5          17  1003000134                       CIBULL   
6          18  1003000134                       CIBULL   
7          19  1003000134                       CIBULL   
8          20  1003000134                       CIBULL   
9          21  1003000134                       CIBULL   

  nppes_provider_first_name nppes_provider_mi nppes_credentials  \
0                    THOMAS                 L              M.D.   
1                    THOMAS                 L              M.D.   
2                    THOMAS                 L              M.D.   
3                    THOMAS                 L              M.D.   
4                    THOMAS                 L              M.D.   
5                    THOMAS                 L              M.D.   
6                    THOMAS                 L              M.D.   
7                    THOMAS                 L              M.D.   
8                    THOMAS                 L              M.D.   
9                    THOMAS                 L              M.D.   

  nppes_provider_gender nppes_entity_code nppes_provider_street1  \
0                     M                 I         2650 RIDGE AVE   
1                     M                 I         2650 RIDGE AVE   
2                     M                 I         2650 RIDGE AVE   
3                     M                 I         2650 RIDGE AVE   
4                     M                 I         2650 RIDGE AVE   
5                     M                 I         2650 RIDGE AVE   
6                     M                 I         2650 RIDGE AVE   
7                     M                 I         2650 RIDGE AVE   
8                     M                 I         2650 RIDGE AVE   
9                     M                 I         2650 RIDGE AVE   

  nppes_provider_street2              ...              hcpcs_code  \
0      EVANSTON HOSPITAL              ...                   88304   
1      EVANSTON HOSPITAL              ...                   88305   
2      EVANSTON HOSPITAL              ...                   88312   
3      EVANSTON HOSPITAL              ...                   88313   
4      EVANSTON HOSPITAL              ...                   88321   
5      EVANSTON HOSPITAL              ...                   88323   
6      EVANSTON HOSPITAL              ...                   88341   
7      EVANSTON HOSPITAL              ...                   88342   
8      EVANSTON HOSPITAL              ...                   88344   
9      EVANSTON HOSPITAL              ...                   88346   

                                   hcpcs_description hcpcs_drug_indicator  \
0  Pathology examination of tissue using a micros...                    N   
1  Pathology examination of tissue using a micros...                    N   
2  Special stained specimen slides to identify or...                    N   
3  Special stained specimen slides to examine tis...                    N   
4         Surgical pathology consultation and report                    N   
5         Surgical pathology consultation and report                    N   
6  Special stained specimen slides to examine tissue                    N   
7   Tissue or cell analysis by immunologic technique                    N   
8  Special stained specimen slides to examine tissue                    N   
9                                Antibody evaluation                    N   

  line_srvc_cnt bene_unique_cnt bene_day_srvc_cnt  \
0         261.0           237.0             240.0   
1        7841.0          4488.0            5664.0   
2         573.0           373.0             387.0   
3         110.0            94.0              96.0   
4   

In [20]:
list(df15)

['Unnamed: 0',
 'npi',
 'nppes_provider_last_org_name',
 'nppes_provider_first_name',
 'nppes_provider_mi',
 'nppes_credentials',
 'nppes_provider_gender',
 'nppes_entity_code',
 'nppes_provider_street1',
 'nppes_provider_street2',
 'nppes_provider_city',
 'nppes_provider_zip',
 'nppes_provider_state',
 'nppes_provider_country',
 'provider_type',
 'medicare_participation_indicator',
 'place_of_service',
 'hcpcs_code',
 'hcpcs_description',
 'hcpcs_drug_indicator',
 'line_srvc_cnt',
 'bene_unique_cnt',
 'bene_day_srvc_cnt',
 'average_Medicare_allowed_amt',
 'average_submitted_chrg_amt',
 'average_Medicare_payment_amt',
 'average_Medicare_standard_amt']

In [13]:
genetic15 = df15[df15['hcpcs_description'].str.contains("genetic", na=False)]

In [16]:
test_type15 = genetic15.groupby('hcpcs_description').size().reset_index(name='count')
test_type15.sort_values(by='count', ascending=False,inplace=True)
test_type15

hcpcs_description  count
8               Microscopic genetic analysis of tumor   4126
7              Microscopic genetic analysis of tissue    714
9              Microscopic genetic examination manual    547
0                        Analysis of genetic material    498
10  Microscopic genetic examination using computer...    383
5        Interpretation and report of genetic testing    187
2             Chromosome analysis for genetic defects     52
3                     DNA testing for genetic defects     13
6        Microscopic genetic analysis of nerve tissue      6
4     Identification of organisms by genetic analysis      2
1                  Chemical test for genetic disorder      1

In [22]:
tumors15 = genetic15[genetic15['hcpcs_description'].str.contains("Microscopic genetic analysis of tumor", na=False)]
tumors15.groupby('provider_type').size()

provider_type
Hematology/Oncology       3
Medical Oncology          1
Pathology              4122
dtype: int64

In [23]:
report15 = genetic15[genetic15['hcpcs_description'].str.contains("Interpretation and report of genetic testing", na=False)]
report15.groupby('provider_type').size()

provider_type
Pathology    187
dtype: int64

In [24]:
tumors15

Unnamed: 0         npi nppes_provider_last_org_name  \
10              22  1003000134                       CIBULL   
312           7250  1003043167                         WANG   
367           8410  1003050303                        CHENG   
368           8411  1003050303                        CHENG   
398           8529  1003051269                     MCQUITTY   
436           9907  1003063397                          LEE   
454          10061  1003064700                           LI   
524          11158  1003072489                      SANCHEZ   
694          13457  1003080797                        SOPHA   
808          18349  1003130832                      KILGORE   
817          18367  1003131038                       JOSEPH   
909          24974  1003800152                         WANG   
1019         26202  1003801812                     GOLDBERG   
1770         42022  1003828757                          CHU   
1848         42364  1003829730                     MORELAND   
1849         42365  1003829730                     MORELAND   
1896         44880  1003837089                         FINE   
2019         48005  1003843640                      LAZENBY   
2074         49759  1003847492                       BRUNER   
2090         49946  1003848052                       MEMOLI   
2158         52061  1003852674                      EPSTEIN   
2279         54678  1003859398                         CHEN   
2380         56105  1003862228                       SHUKLA   
2397         56634  1003863069                          KIM   
2461         57716  1003865197                        KRANE   
2819         62895  1003874686                     VDOVENKO   
2901         63178  1003875329                     MONTEIRO   
2917         63983  1003876624                        MEYER   
3132         67177  1003882036                       GAINES   
3510         69811  1003886557                     SILVEIRA   
...            ...         ...                          ...   
456498     9395541  1982891016                           LY   
456745     9412993  1992701379                      STROBEL   
456763     9413870  1992702518                      SCHARRE   
456764     9413871  1992702518                      SCHARRE   
457007     9423773  1992724355                       SPEAKS   
457243     9427887  1992734503                       COOPER   
457278     9428582  1992735740                     MCDOWELL   
457279     9428583  1992735740                     MCDOWELL   
457304     9428857  1992736268                        DAVIS   
457724     9434158  1992746978                       FANDEL   
458078     9443398  1992764138                      MCCLURE   
458100     9443788  1992764880                        FIDDA   
458101     9443789  1992764880                        FIDDA   
458140     9445160  1992767487                           LU   
458306     9448446  1992772891                        FLYNN   
458307     9448447  1992772891                        FLYNN   
458409     9448699  1992773196                       BUTLER   
458600     9451322  1992777593                           PU   
458734     9454882  1992783617                     LAWNICKI   
458763     9454993  1992783831                     PONIECKA   
459431     9470741  1992818967                      BOHLING   
459549     9473574  1992845846                   FULENWIDER   
459598     9475754  1992861207                        NABER   
459642     9476346  1992865067                        BRODY   
459873     9479892  1992882807                         THOR   
460014     9483202  1992898688                    DUNCAVAGE   
460157     9486248  1992913032                        DEWAN   
460279     9487973  1992925861                       MURALI   
460407     9490958  1992949309                         GILL   
460488     9492915  1992962849                         CHAN   

       nppes_provider_first_name nppes_provider_mi nppes_credentials  \
10                        THO

In [58]:
# List the addresses of tumor claims

tumors15.nppes_provider_street1

10                    2650 RIDGE AVE
312                   251 E HURON ST
367                3426 BRAEBURN CIR
368                3426 BRAEBURN CIR
398             3600 NW SAMARITAN DR
436            660 N WESTMORELAND RD
454                   2160 S 1ST AVE
524           601 WEST SECOND STREET
694                   600 N WOLFE ST
808               1959 NE PACIFIC ST
817               535 E CRESCENT AVE
909                     1638 OWEN DR
1019                   901 MACARTHUR
1770                 90 S BEDFORD RD
1848           2904 WESTCORP BLVD SW
1849           2904 WESTCORP BLVD SW
1896                   300 HALKET ST
2019           619 19TH STREET SOUTH
2074               5301 VIRGINIA WAY
2090             1 MEDICAL CENTER DR
2158                    1 HOAG DRIVE
2279                  2411 HOLMES ST
2380                 1007 LINCOLNWAY
2397                1500 E DUARTE RD
2461           75 FRANCIS ST AMORY 3
2819                    100 GRAND ST
2901                    655 W 8TH ST
2

## LOAD THE 215 RESEARCH CLINICS, TO MATCH ADDRESSES



In [46]:
research = pd.read_csv(datadir + '215_research_clinics.csv')
research.shape

(215, 24)

In [49]:
# convert two address columns to Series
research_mailing_address = pd.Series(research['Mailing Address'])
research_practice_address = pd.Series(research['Practice Address'])

In [56]:
# perform matching
(
    tumors15.nppes_provider_street1.isin(research_mailing_address) |
    tumors15.nppes_provider_street1.isin(research_practice_address)
).value_counts()

False    4114
True       12
Name: nppes_provider_street1, dtype: int64

So of the **4126** tumor tests, 12 came from research clinics

In [60]:
(
    genetic15.nppes_provider_street1.isin(research_mailing_address) |
    genetic15.nppes_provider_street1.isin(research_practice_address)
).value_counts()

False    6512
True       17
Name: nppes_provider_street1, dtype: int64

# LOAD THE 25,659 Oncology clinics, to match addresses

In [91]:
onclinics = pd.read_csv(datadir + '25K-oncology-deduped.csv')
onclinics.shape

(10190, 3)

In [92]:
list(onclinics)

['NPI', 'ADDRESS', 'Healthcare Provider Taxonomy Code_1']

In [95]:
onclinic_address = pd.Series(onclinics.ADDRESS)

In [96]:
onclinic_address

0                                             1450 8TH AVE
1                           80 HUMPHREYS CENTER DR STE 330
2                                         1020 FRANKLIN ST
3                                         210 SE 136TH AVE
4                                        19260 SW 65TH AVE
5                                  7525 GREENWAY CENTER DR
6                                      262 DANNY THOMAS PL
7                                          5400 N OAK TRFY
8                                    1850 W ARLINGTON BLVD
9                                            2100 ERWIN RD
10                                      1515 HOLCOMBE BLVD
11                   ST. JUDE CHILDREN'S RESEARCH HOSPITAL
12                                     111 MICHIGAN AVE NW
13                                      15990 MEDICAL DR S
14                                       265 N BROADWAY ST
15                                         4610 BOBBITT DR
16                                                # L-35

In [98]:
genetic15.nppes_provider_street1.isin(onclinic_address).value_counts()

False    3293
True     3236
Name: nppes_provider_street1, dtype: int64

In [101]:
3293+3236

6529

In [75]:
# convert two address columns to Series
onclinics_mailing_address = pd.Series(onclinics['Mailing Address'])
onclinics_practice_address = pd.Series(onclinics['Practice Address'])

KeyError: 'Mailing Address'

In [69]:
# perform matching
(
    tumors15.nppes_provider_street1.isin(onclinics_mailing_address) |
    tumors15.nppes_provider_street1.isin(onclinics_practice_address)
).value_counts()

True     2174
False    1952
Name: nppes_provider_street1, dtype: int64

In [70]:
list(onclinics)

['Unnamed: 0',
 'NPI',
 'Entity Type Code',
 'Provider Organization Name (Legal Business Name)',
 'Mailing Address',
 'Practice Address',
 'Healthcare Provider Taxonomy Code_1',
 'Healthcare Provider Taxonomy Code_2',
 'Healthcare Provider Taxonomy Code_3',
 'Healthcare Provider Taxonomy Code_4',
 'Healthcare Provider Taxonomy Code_5',
 'Healthcare Provider Taxonomy Code_6',
 'Healthcare Provider Taxonomy Code_7',
 'Healthcare Provider Taxonomy Code_8',
 'Healthcare Provider Taxonomy Code_9',
 'Healthcare Provider Taxonomy Code_10',
 'Healthcare Provider Taxonomy Code_11',
 'Healthcare Provider Taxonomy Code_12',
 'Healthcare Provider Taxonomy Code_13',
 'Healthcare Provider Taxonomy Code_14',
 'Healthcare Provider Taxonomy Code_15',
 'Is Organization Subpart',
 'Parent Organization LBN',
 'Parent Organization TIN']

In [71]:
1952+2174

4126

In [51]:
tumors15.nppes_provider_street1.isin(research_practice_address).value_counts()

False    4114
True       12
Name: nppes_provider_street1, dtype: int64

In [32]:
peachtree = tumors15.nppes_provider_street1.str.contains("1968 PEACHTREE RD NW")
tumors15[peachtree]

Unnamed: 0         npi nppes_provider_last_org_name  \
67251      1380956  1144298217                       WALKER   
95171      1963499  1205881802                       SNYDER   
181672     3748592  1396749164                        CHENG   
182517     3765382  1396787065                     RUFFORNY   
230526     4742313  1497930010                     CHANDLER   
458409     9448699  1992773196                       BUTLER   

       nppes_provider_first_name nppes_provider_mi nppes_credentials  \
67251                      BRUCE                 F                MD   
95171                     TAMELA                 M                MD   
181672                  JEANETTE                 D                MD   
182517                     IRINA               NaN                MD   
230526                 KATHERINE                 L              M.D.   
458409                   BRADLEY                 S                MD   

       nppes_provider_gender nppes_entity_code nppes_provider_street1  \
67251                      M                 I   1968 PEACHTREE RD NW   
95171                      F                 I   1968 PEACHTREE RD NW   
181672                     F                 I   1968 PEACHTREE RD NW   
182517                     F                 I   1968 PEACHTREE RD NW   
230526                     F                 I   1968 PEACHTREE RD NW   
458409                     M                 I   1968 PEACHTREE RD NW   

                    nppes_provider_street2              ...               \
67251                       PATHOLOGY DEPT              ...                
95171                       PATHOLOGY DEPT              ...                
181672  DEPARTMENT OF PATHOLOGY, 4TH FLOOR              ...                
182517                      PATHOLOGY DEPT              ...                
230526              77 BUILDING, 4TH FLOOR              ...                
458409                                 NaN              ...                

       hcpcs_code                      hcpcs_description hcpcs_drug_indicator  \
67251       88361  Microscopic genetic analysis of tumor                    N   
95171       88361  Microscopic genetic analysis of tumor                    N   
181672      88361  Microscopic genetic analysis of tumor                    N   
182517      88361  Microscopic genetic analysis of tumor                    N   
230526      88361  Microscopic genetic analysis of tumor                    N   
458409      88361  Microscopic genetic analysis of tumor                    N   

       line_srvc_cnt bene_unique_cnt bene_day_srvc_cnt  \
67251           41.0            13.0              13.0   
95171           66.0            18.0              19.0   
181672          53.0            15.0              15.0   
182517         107.0            30.0              30.0   
230526          45.0            14.0              14.0   
458409          67.0            18.0              18.0   

       average_Medicare_allowed_amt average_submitted_chrg_amt  \
67251                     59.410732                      250.0   
95171                     60.312727                      250.0   
181672                    60.230566                      250.0   
182517                    59.945701                      250.0   
230526                    57.773333                      250.0   
458409                    59.702687                      250.0   

       average_Medicare_payment_amt average_Medicare_standard_amt  
67251                     46.578049                     39.249268  
95171                     46.955000                     44.103030  
181672                    47.220943                     38.399811  
182517                    45.920467                     38.483271  
230526                    44.233333                     37.075333  
458409                    45.087015                     33.201642  

[6 rows x 27 columns]

In [36]:
tumors15.nppes_provider_street1.isin(["1968 PEACHTREE RD NW","1 MEDICAL CENTER DR"]).value_counts()

False    4111
True       15
Name: nppes_provider_street1, dtype: int64

In [5]:
entities = df15.groupby('nppes_entity_code').size().reset_index(name='count')
entities.sort_values(by='count', ascending=False,inplace=True)
entities

nppes_entity_code   count
0                 I  460649

In [6]:
providers = df15.groupby('provider_type').size().reset_index(name='count')
providers.sort_values(by='count', ascending=False,inplace=True)
providers

provider_type   count
1     Hematology/Oncology  169136
3               Pathology  149828
4      Radiation Oncology   75950
2        Medical Oncology   52443
0  Gynecological/Oncology    7901
5       Surgical Oncology    5391

In [8]:
# genetic12 = df12[df12['HCPCS_DESCRIPTION'].str.contains("genetic", na=False)]

In [9]:
# gsheetname = 'specialist_genetic_tests'
# book = client.open(gsheetname)
# sheet_name ="Subset"
# sheet = gsheet.worksheet(sheet_name)
# df = pd.DataFrame(sheet.get_all_records())
# df

In [10]:
# provider13_sheet = book.add_worksheet(title="2013_provider_counts", rows="100", cols="20")

In [11]:
# for row in provider13.iterrows():
#     provider13_sheet.append_row(row[1].tolist())